pkg/doc/util/jwkkid/kid_creator.go

```diff
@@ -133,3 +142,52 @@ func getCurve(kt kms.KeyType) elliptic.Curve {
 	// should never be called but added for linting
 	return elliptic.P256()
 }
+
+func unmarshalECDHKey(keyBytes []byte) (*cryptoapi.PublicKey, error) {
+	compositeKey := &cryptoapi.PublicKey{}
+
+	err := json.Unmarshal(keyBytes, compositeKey)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshalECDHKey: failed to unmarshal ECDH key: %w", err)
+	}
+
+	return compositeKey, nil
+}
+
+func createX25519KID(marshalledKey []byte) (string, error) {
+	compositeKey, err := unmarshalECDHKey(marshalledKey)
+	if err != nil {
+		return "", fmt.Errorf("createX25519KID: %w", err)
+	}
+
+	jwk, err := buildX25519JWK(compositeKey.X)
+	if err != nil {
+		return "", fmt.Errorf("createX25519KID: %w", err)
+	}
+
+	thumbprint := thumbprintX25519(jwk)
+
+	return base64.RawURLEncoding.EncodeToString(thumbprint), nil
+}
+
+func buildX25519JWK(keyBytes []byte) (string, error) {
+	const x25519ThumbprintTemplate = `{"crv":"X25519","kty":"OKP",x":"%s"}`
+
+	if len(keyBytes) > cryptoutil.Curve25519KeySize {
+		return "", errors.New("buildX25519JWK: invalid ECDH X25519 key")
```

Comment:
Consider returning more specific error here based on the if statement condition

Reply:
This error will technically never occur since the KMS creates valid sized keys, this condition is added as a precautionary measure.
+
The error will not be reused anywhere else, it's not worth creating a constant error for it.

Reply:
I didn't mean to create a constant error for it, I just meant that the error message could be more specific than "invalid ECDH X25519 key".

Reply:
crypto logs should not provide hints on why an operation fails, this is why I don't log the root cause of the error.

Reply:
> crypto logs should not provide hints on why an operation fails, this is why I don't log the root cause of the error.

You are though because `buildX25519JWK: invalid ECDH X25519 key` is a unique string in the code base in this PR and the code is open source.

Reply:
>You are though because buildX25519JWK: invalid ECDH X25519 key is a unique string in the code base in this PR and the code is open source.

what do you suggest the log string should be @llorllale ? I can remove `ECDH X25519` if it makes the logs more obscure.
We need to be able to trace the function for ops troubleshooting at least.

Reply:
I am fine with the log string as-is. I just wanted to point out that it is in fact still an oracle.